barrasso/barrassoic3token.sol

```diff
@@ -0,0 +1,94 @@
+pragma solidity ^0.4.8;
+
+contract IC3Token {
+    string public constant symbol = "IC3";
+    string public constant name = "IC3 2017 Bootcamp Token";
+    uint8 public constant decimals = 18;
+    uint256 _totalSupply;
+
+    // balances for each account
+    mapping(address => uint256) balances;
+    
+    // approve the transfer of an amount to another account
+    mapping(address => mapping (address => uint256)) allowed;
+    
+    // transfer and approval events
+    event Transfer(address indexed _from, address indexed _to, uint256 _value);
+    event Approval(address indexed _owner, address indexed _spender, uint256 _value);
+    
+    // constructor
+    function IC3Token() {
+        _totalSupply = 0;
+    }
+    
+    function totalSupply() constant returns (uint256 totalSupply) {
+        totalSupply = _totalSupply;
+    }
+    
+    // what is balance of a particular account
+    function balanceOf(address _owner) constant returns (uint256 balance) {
+        return balances[_owner];
+    }
+    
+    // transfer the balance from one owner's account to another account
+    function transfer(address _to, uint256 _amount) returns (bool success) {
+        if (balances[msg.sender] >= _amount
+            && _amount > 0 
```

Comment:
This isn't necessary to check, 0 amounts should return true (this is not clear in the spec so you didn't do anything wrong :)).